tests/ga/test_extension.py

```diff
@@ -1327,6 +1327,44 @@ def test__extension_upgrade_failure_when_prev_version_disable_fails(self, patch_
         # On the next iteration, update should not be retried
         self._assert_handler_status(protocol.report_vm_status, "NotReady", expected_ext_count=0, version="1.0.1")
 
+    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_disable_command')
+    def test__old_handler_reports_failure_on_disable_fail_on_update(self, patch_get_disable_command, *args):
+        test_data = WireProtocolData(DATA_FILE_EXT_SINGLE)
+        exthandlers_handler, protocol = self._create_mock(test_data, *args)
+        old_version, new_version = "1.0.0", "1.0.1"
+
+        # Ensure initial install and enable is successful
+        exthandlers_handler.run()
+
+        self.assertEqual(0, patch_get_disable_command.call_count)
+        self._assert_handler_status(protocol.report_vm_status, "Ready", expected_ext_count=1, version=old_version)
+        self._assert_ext_status(protocol.report_ext_status, "success", 0)
+
+        # Next incarnation, update version
+        test_data.goal_state = test_data.goal_state.replace("<Incarnation>1<", "<Incarnation>2<")
+        test_data.ext_conf = test_data.ext_conf.replace('version="%s"' % old_version, 'version="%s"' % new_version)
+        test_data.manifest = test_data.manifest.replace(old_version, new_version)
+
+        with patch.object(ExtHandlerInstance, "report_event", autospec=True) as patch_report_event:
+            # Disable of the old extn fails
+            patch_get_disable_command.return_value = "exit 1"
+            exthandlers_handler.run()  # Download the new update the first time, and then we patch the download method.
+            self.assertEqual(1, patch_get_disable_command.call_count)
+
+            new_version_args, new_version_kwargs = patch_report_event.call_args
```

Comment:
Wouldn't `report_event` get called from the old version first, from the failed disable command?

Reply:
yes you're correct, the report event gets called 3 times in this test scenario run. 
`patch_report_event.call_args` This property saves the last call only (`patch_report_event.call_args_list` saves all the times this method was called). In our case the last call to report_event gets called from the new version since the error is propagated back to the calling method. 

`new_version_args, new_version_kwargs = patch_report_event.call_args` 
This is getting the data for the last call. I'm using it to check that `log_event` is false for that one as we've already logged it earlier in the 2nd call to report_event